rudder-core/src/main/scala/com/normation/rudder/services/queries/LdapQueryProcessor.scala

```diff
@@ -730,38 +711,36 @@ class InternalLDAPQueryProcessor(
    */
   private def normalize(query:Query) : Box[LDAPNodeQuery] = {
 
-    //validate that we knows the requested object type
-    if(!objectTypes.isDefinedAt(query.returnType.value))
-      return Failure("The requested type '%s' is not know".format(query.returnType))
-
-    //validate that for each object type in criteria, we know it
-    query.criteria foreach { cl =>
-      if(!objectTypes.isDefinedAt(cl.objectType.objectType))
-        return Failure("The object type '%s' is not know in criteria %s".format(cl.objectType.objectType,cl))
+    def groupFilterByLdapRequest() = {
+      //validate that we knows the requested object type
+      if (objectTypes.isDefinedAt(query.returnType.value)) {
+        (sequence(query.criteria) {
+           case crit@CriterionLine(ot,a,comp,value) => {
+            val objectType = ot.objectType
```

Comment:
The first "group by" was risky because we were making a group by then a map that changes the keys which could make a loss of data ( Would only keep the last thing with the same 'tpe', which does not appear now but it could and it would be veryyyy surprising) 

Here I compute all data we need in one go before grouping them, we produce the same data that we would produce